lib/console.js

```diff
@@ -162,6 +162,25 @@ Console.prototype.dir = function dir(object, options) {
 };
 
 
+Console.prototype.dirxml = function dirxml(...data) {
+  const optionProps = ['showHidden', 'depth', 'colors'],
+    maybeOptions = Object.getOwnPropertyNames(data.slice(-1)),
+    isOption = maybeOptions.some((p) => optionProps.indexOf(p) !== -1);
```

Comment:
Is the last argument supposed to be an options object? The spec doesn't seem to call for that, and it makes for a slightly awkward UX, IMO. What if we want to display an object that has one of those magic props?

Reply:
I don't get why this is here. The spec doesn't allow to specify options for `dirxml`. The fact that there has to be this ugly code to detect it should indicate that it should not exist in the first place.

Reply:
That's kind of hard said like that.

Anyway, I thought about the possibility of having the last object having precisely one of those options, but I thought that it would be better to allow passing options if need be, just like `console.dir()` and that the risk would be quite minimal.
Plus, it feels a bit weird to have `dir` accept options but take only one object, and `dirxml` not accepting options but taking a varying number of object, but maybe that's just me.
And it does remove the possibility for custom inspect, and so xml redering.

I'll remove it for now anyway, since it seems to be a bad idea, it's indeed akward.

Reply:
`dir` explicitly lists `options` as something that it accepts in the spec: https://console.spec.whatwg.org/#dir

When working on features which have a spec, decisions should always be informed by the behaviour defined there.

Reply:
Also, since the `console` spec is an early draft that's classified as a Living Standard, there's always the opportunity for you to go back and try to suggest improvement at the source. But Nodejs shouldn't be the origin for that divergence since it just creates confusion for developers.

Reply:
BTW, the `console.dir` spec was in fact amended to account for the possibility of having an options object because of Node.js (see https://github.com/whatwg/console/issues/79). However, I agree with @apapirovski that

> Node.js shouldn't be the origin for that divergence since it just creates confusion for developers.

-- not if we can help it anyway, which we couldn't with `console.dir` w/o breaking backwards compatibility.

Reply:
Yep, definitely agree and good to have a bit of background on that! Thanks @TimothyGu.

Reply:
Thanks for the background indeed!

However, I don't know if `dir` is widely used, but I don't know either if it would be worth it to go and ask if the standard could be adapted.
Any opinion on this?

Still dropping this while the standard is as its current state anyway, it does make sense, thank you all!

Reply:
The problem in this particular case is that `dirxml` accepts a rest parameter which always has to be final, it's similar in that to `log` and most other `console` functions.

There would need to either be a method to set general settings for all `dirxml` calls or some options object that is exposed on the console, or some sort of a `Symbol` (similar to like `@@toStringTag`) that gets declared on the object that decides the formatting preferences.

Reply:
Indeed that's a problem, that's what had me attempt (poorly) to parse the last object in the list. I don't mind attempting again if the standard does evolve though. If it evolves and if it's worth it. I don't know if it's the case though.